pkg/registries/ecr/ecr.go

```diff
@@ -84,8 +89,17 @@ func sanitizeConfiguration(ecr *storage.ECRConfig) error {
 	return errorList.ToError()
 }
 
+func (e *ecr) setLastRefreshAttempt() {
+	e.lastRefreshAttempt = time.Now()
+}
+
 func (e *ecr) refreshDockerClient() error {
-	if e.expiryTime.After(time.Now()) {
+	e.lock.Lock()
+	defer e.lock.Unlock()
+
+	defer e.setLastRefreshAttempt()
```

Comment:
I fear setting this every single time may mean `time.Since(e.lastRefreshAttempt) < refreshAttemptInterval` is potentially always true

Reply:
oh yeah I want to move this to below the check. good call